hw/syn/tools/dc/parse-syn-report.py

```diff
@@ -593,12 +593,16 @@ def main():
                    for_json=True,
                    use_decimal=True)
 
+    # helper function
+    def _getlen(x):
+        return len(x) if x is not None else 0
+
     # return nonzero status if any warnings or errors are present
     # lint infos do not count as failures
-    nr_errors = (len(results["messages"]["flow_errors"]) +
-                 len(results["messages"]["analyze_errors"]) +
-                 len(results["messages"]["elab_errors"]) +
-                 len(results["messages"]["compile_errors"]))
+    nr_errors = (_getlen(results["messages"]["flow_errors"]) +
```

Comment:
I normally try to avoid assign multiple types to a variable. If `flow_errors`, `analyze_errors`,etc are intended to be list, could you change 390-393 to assign empty list rather than None?

Reply:
Initially this was handled the way you describe.

After adding support for partial builds however (e.g. elab-only), some of these may now be `None` when no results are available for a specific step since it has not been run. 
(Note: an empty list signifies that results available but no messages have been found in this category).

Another way of handling this would be to add more variables to the `results.hjson` that indicate whether results are available for a specific stage.

Reply:
I thought somehow `termination_stage` is stored into the result data structure but looks like not. Then yep. unless you add a new variable, this is good. Still I prefer adding the stage or something similar to the structure :)

Reply:
Yeah a cleaner way of doing it is to make `SynCfg.py` aware of the stages - but I wanted to keep most of the changes on the result parser and Hjson cfg side for now so that we don't have to touch too much Dvsim core code. 

But at some point I will have to do a refactor pass of the result parsing for `SynCfg.py` similar to `LintCfg.py` (see https://github.com/lowRISC/opentitan/issues/9079).